### PR TITLE
Update AOTS tests and fix some failures

### PR DESCRIPTION
**Pull request by khaledhosny**

### CODE REVIEW

File: util/main-font-text.hh [hunk 1]
```diff
@@ -47,7 +47,7 @@ locale_to_utf8 (char *s)
   return t;
 }
 
-template <typename consumer_t, int default_font_size, int subpixel_bits>
+template <typename consumer_t, int default_font_size, int subpixel_bits, bool oneline = false>
```
Comment (behdad) [inline util/main-font-text.hh:50]: How about adding a `char eol = '\n'` instead?

File: util/options.cc [hunk 2]
```diff
@@ -258,12 +258,21 @@ parse_features (const char *name G_GNUC_UNUSED,
...
+  size_t l = strlen (s);
   char *p;
 
   shape_opts->num_features = 0;
   g_free (shape_opts->features);
   shape_opts->features = nullptr;
 
+  /* if the string is quoted strip the quotes */
+  if ((s[0] == '\"' && s[l - 1] == '\"') ||
+      (s[0] == '\'' && s[l - 1] == '\''))
```
Comment (behdad) [inline util/options.cc:270]: Slightly nicer would be to check that `s[0] == s[l - 1]` first.

**Comment by behdad** [comment]
Thanks Khaled. LGTM.

File: util/main-font-text.hh (hunk 1 above)
Comment (khaledhosny) [reply util/main-font-text.hh]: Done.

File: util/options.cc (hunk 2 above)
Comment (khaledhosny) [reply util/options.cc]: Done.

File: util/options.hh [hunk 3]
```diff
@@ -548,7 +548,7 @@ struct text_options_t : option_group_t
 		   "Only one of text and text-file can be set");
   }
 
-  const char *get_line (unsigned int *len);
+  const char *get_line (unsigned int *len, char eol);
```
Comment (behdad) [inline util/options.hh:551]: Maybe add the default `\n` here?

Comment (behdad) [reply util/options.hh:551]: Did it myself. Thanks agian.

Comment (khaledhosny) [reply util/options.hh:551]: I considered that, but found it is called only in one place. But thinking again, I agree, it is more elegant/self documenting.

**Pull request merged**